File: java/iroha_android/src/main/java/org/hyperledger/iroha/android/telemetry/DeviceProfileProvider.java
package org.hyperledger.iroha.android.telemetry;

import java.util.Optional;

/** Provider that supplies the device-profile bucket for telemetry emissions. */
public interface DeviceProfileProvider {

  /** Returns the current device profile when available. */
  Optional<DeviceProfile> snapshot();

  /** Returns a no-op provider. */
  static DeviceProfileProvider disabled() {
    return Optional::empty;
  }
}

File: .taira-source-c80bf4d85a97/kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/telemetry/DeviceProfileProvider.kt
package org.hyperledger.iroha.sdk.telemetry

import java.util.Optional

/** Provider that supplies the device-profile bucket for telemetry emissions. */
fun interface DeviceProfileProvider {

    /** Returns the current device profile when available. */
    fun snapshot(): Optional<DeviceProfile>

    companion object {
        /** Returns a no-op provider. */
        @JvmStatic
        fun disabled(): DeviceProfileProvider = DeviceProfileProvider { Optional.empty() }
    }
}
